### PR TITLE
Provide helpers for sanity checking.

### PR DESCRIPTION
**Pull request by ghost**
Many functions throughout lib/ repeat sanity checks on $name and $dataset. The functions __require_name and __require_dataset are intended to replace these repetitive tasks.

Update __set_jail_prop to use the new sanity checking helpers.

### CODE REVIEW

**Comment by freqlabs** [comment]
Meant to hold off on that second commit, but oops there it is now.

File: lib/ioc-common [hunk 1]
```diff
@@ -10,17 +10,25 @@ __export_props () {
...
-    if [ -z "$_name" ] || [ -z "$_property" ] ; then
-        echo "  ERROR: Missing property or UUID!"
+    _name="$1"
+
+    if [ -z "$_name" ] ; then
+        echo "  ERROR: Missing UUID!"
         exit 1
     fi
 
+    echo "$_name"
```
Comment (skarekrow) [inline lib/ioc-common:24]: Is there a reason you have this here? Or is that just debug you forgot in?

File: lib/ioc-common [hunk 2]
```diff
@@ -33,6 +41,22 @@ __set_jail_prop () {
         exit 1
     fi
 
+    echo "$_dataset"
```
Comment (skarekrow) [inline lib/ioc-common:44]: Same as above.

**Comment by freqlabs** [comment]
That's effectively the return statement for the functions. They don't modify variables outside their scope, instead they just echo their output. That's why they are used as _name="$(__require_name $1)" and _dataset="$(__require_dataset $_name)". The helpers perform the necessary checks, returning the value to be stored if the checks pass, otherwise displaying an error message and exiting.

**Comment by pannon** [comment]
@freqlabs definitely supportive on the idea. Just need some time to test.

**Pull request closed without merging**